services/hipchat/service.go

```diff
@@ -0,0 +1,93 @@
+package hipchat
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+
+	"github.com/influxdb/kapacitor"
+)
+
+type Service struct {
+	room   string
+	token  string
+	url    string
+	global bool
+	logger *log.Logger
+}
+
+func NewService(c Config, l *log.Logger) *Service {
+	return &Service{
+		room:   c.Room,
+		token:  c.Token,
+		url:    c.URL,
+		global: c.Global,
+		logger: l,
+	}
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+func (s *Service) Global() bool {
+	return s.global
+}
+
+func (s *Service) Alert(room, token, message string, level kapacitor.AlertLevel) error {
+
+	//Generate HipChat API Url including room and authentication token
+	if room == "" {
+		room = s.room
+	}
+	if token == "" {
+		token = s.token
+	}
+
+	url := s.url + "/" + room + "/notification?auth_token=" + token
```

Comment:
Should we URL escape the `room` and `token` vars so the config doesn't have to do things like `Test%20Room`? See the answer to this SO question. It has a good example of how to safely craft the URL.  http://stackoverflow.com/questions/13820280/encode-decode-urls